File: app/src/main/java/com/dumuzeyn/mp3player/TrackOrigin.java
package com.dumuzeyn.mp3player;

/** Stable ownership data for a track discovered inside a SAF tree. */
final class TrackOrigin {
    final String trackId;
    final String sourceId;
    final String documentId;
    final String identityKey;

    TrackOrigin(String trackId, String sourceId, String documentId, String identityKey) {
        this.trackId = trackId;
        this.sourceId = sourceId;
        this.documentId = documentId;
        this.identityKey = identityKey;
    }

    static String identity(String sourceId, String documentId) {
        return "document:" + sourceId + ':' + documentId;
    }

    static String uriIdentity(String uri) {
        return "uri:" + (uri == null ? "" : uri);
    }
}

File: app/src/main/java/com/dumuzeyn/mp3player/TrackOrigin.kt
package com.dumuzeyn.mp3player

/** Stable ownership data for a track discovered inside a SAF tree. */
class TrackOrigin(
    @JvmField val trackId: String,
    @JvmField val sourceId: String,
    @JvmField val documentId: String,
    @JvmField val identityKey: String,
) {
    companion object {
        @JvmStatic
        fun identity(sourceId: String, documentId: String): String =
            "document:$sourceId:$documentId"

        @JvmStatic
        fun uriIdentity(uri: String?): String = "uri:${uri.orEmpty()}"
    }
}
